Digital Ocean integration test for digital_ocean_tag

kbreit opened the pull request:
##### SUMMARY
None of the Digital Ocean modules have integration tests. This is an integration test for the tag module. There's very little around validation, but it at least catches obvious errors.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
digital_ocean_tag

##### ADDITIONAL INFORMATION
Credential information should go into the `integration_config.yml` file.

kbreit committed:
Add Digital Ocean integration test template

- Actual integration tests will come shortly

kbreit committed:
Add digital_ocean-tag test

kbreit committed:
Add integration test for creating and deleting tags

kbreit committed:
Remove cloud confiugration file

kbreit commented:
@gundalow This is a duplicate PR for https://github.com/ansible/ansible/pull/49316. Should I use this one instead?

gundalow commented:
@kbreit use whichever PR works best for you.

kbreit commented:
@gundalow I'd like to use this one. Please review.

gundalow commented on test/integration/targets/digital_ocean_tag/tasks/main.yml:
```diff
@@ -0,0 +1,29 @@
+    that:
+      - create_tag.changed == True
+      - create_tag.data.tag.name == "integration-test"
+
```
To confirm idempotent, create again and assert no changed.

Same for delete.

kbreit replied:
Added test for idempotency

gundalow commented on test/integration/targets/digital_ocean_tag/tasks/main.yml:
```diff
@@ -0,0 +1,29 @@
+
+- assert:
+    that:
+      - delete_tag.changed == True
```
Missing newline.

kbreit replied:
Looks like the module has a idempotency bug so I can't make this integration test fully work. #51107

kbreit replied:
Delete isn't idempotent.

dagwieers replied:
If delete is not idempotent and we can/will not fix this now, could you add a NOTE or a FIXME to the integration test to indicate this ?

```yaml
      # FIXME: Deleting a tag is not idempotent
      #- delete_tag_idempotent.changed == False
```

gundalow commented on test/integration/targets/digital_ocean_tag/aliases:
```diff
@@ -0,0 +1,2 @@
+digital_ocean
+unsupported
```
Missing newline.

kbreit replied:
Fixed

kbreit committed:
Enhancements for integration test for idempotency
- Added new lines per review
- Added proper tests for idempotency checks
- Actual module is broken so deletion test doesn't work

kbreit committed:
Add check for do_api_key and instruct if it doesn't work.

kbreit committed:
Fix indentation

kbreit committed:
Remove idempotency test for nonidempotency action

kbreit commented:
@gundalow Please give this another look but I think we're ready to go on this one.

BondAnthony commented:
LGTM

shipit

kbreit commented:
bot_status

dagwieers reviewed (approved):
LGTM

kbreit committed:
Add notes for delete idempotency
